src/DotVVM.Framework/Routing/DotvvmRouteTable.cs

```diff
@@ -131,6 +135,49 @@ public void Add(string routeName, string? url, Func<IServiceProvider, IDotvvmPre
             Add(group?.RouteNamePrefix + routeName, new DotvvmRoute(CombinePath(group?.UrlPrefix, url), group?.VirtualPathPrefix ?? "", defaultValues, presenterFactory ?? GetDefaultPresenter, configuration));
         }
 
+        /// <summary>
+        /// Adds the specified URL redirection entry.
+        /// </summary>
+        /// <param name="routeName">Name of the redirection.</param>
+        /// <param name="urlPattern">URL pattern to redirect from.</param>
+        /// <param name="urlProvider">URL provider to obtain context-based redirection targets.</param>
+        public void AddUrlRedirection(string routeName, string urlPattern, Func<IDotvvmRequestContext, string> urlProvider, object? defaultValues = null, bool permanent = false)
+        {
+            ThrowIfFrozen();
+            Func<IServiceProvider, IDotvvmPresenter> presenterFactory = (serviceProvider) => GetRedirectionPresenter((context) =>
```

Comment:
```suggestion
            IServiceProvider presenterFactory(IDotvvmPresenter serviceProvider) => GetRedirectionPresenter((context) =>
```

You can use a local function, that is a bit less verbose.